Sources/NIOFileSystem/Array+FileSystem.swift

```diff
@@ -35,11 +36,26 @@ extension Array where Element == UInt8 {
         self = Self(buffer: byteBuffer)
     }
 
+    /// Reads the contents of the file at the path.
+    ///
+    /// - Parameters:
+    ///   - path: The path of the file to read.
+    ///   - maximumSizeAllowed: The maximum size of file which can be read, in bytes, as a ``ByteCount``.
+    ///   - fileSystem: The ``FileSystemProtocol`` instance to use to read the file.
+    public init(
+        contentsOf path: NIOFilePath,
+        maximumSizeAllowed: ByteCount,
+        fileSystem: some FileSystemProtocol
+    ) async throws {
+        try await self.init(contentsOf: path.underlying, maximumSizeAllowed: maximumSizeAllowed, fileSystem: fileSystem)
```

Comment:
The `NIOFilePath` variant should be the canonical version of the function and the `FilePath` version should call through to it. You'll need to make this change in a bunch of places.